Timer merge

icristescu opened the pull request:
Rebased over PR #100.
 
With the results:
```
Fill in random order
index: 2.824067 micros/op; 	 354099.247645 op/s; 	 15.934466 MB/s; 	 total time = 28.240670s.
	write amplification in bytes = 10.505701; in nb of writes = 0.000587;
	merge time = 23.098916; percentage = 0.817931;
```

pascutto commented:
If you want to call it `percentage` it's probably better to have it multiplied by 100 before.

CraigFe commented:
Metrics already has a timing mechanism in the form of the `run` function, which seems more flexible in the sense that it allows the reporter to determine the time operation used. Is there a particular reason for not using that here?

icristescu commented:
> Metrics already has a timing mechanism in the form of the `run` function, which seems more flexible in the sense that it allows the reporter to determine the time operation used. Is there a particular reason for not using that here?

We removed metrics from the main code base, and moved it to the benchs only, so we can't call metrics from the merge function.

icristescu commented:
Not pertinent anymore, as merges are now done in separate threads.
